src/Symfony/Component/ErrorHandler/ErrorHandler.php

```diff
@@ -640,7 +640,7 @@ public static function handleFatalError(array $error = null): void
      */
     private function renderException(\Throwable $exception): void
     {
-        $renderer = \in_array(\PHP_SAPI, ['cli', 'phpdbg'], true) ? new CliErrorRenderer() : new HtmlErrorRenderer($this->debug);
+        $renderer = \in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true) ? new CliErrorRenderer() : new HtmlErrorRenderer($this->debug);
```

Comment:
this change looks weird to me. In the case of frankenphp, it behaves like a web server, not as a CLI (or would this report as a different value than `embed` ?)

Reply:
FrankenPHP doesn't use the `embed` SAPI in web mode, and in CLI mode... it uses the embed SAPI but changes its name to  `cli`, which is a hack... but for interoperability with existing code like this :-D

`embed` is documented and usually used as a very simple SAPI to execute PHP scripts in C/C++ programs. AFAIK it is never used in web context. IMHO defaulting to `cli` for it is safer (and can be override). Anyway, it's an edge case, I'm fine with not supporting it at all too.

Reply:
We already have a few spots where we consider embed as cli. Note that I didn't add embed everywhere cli is listed, especially when opcache is considered. The reason is that I don't have enough background on the topic to decide if we should change.

Reply:
opcache doesn't work with embed: https://github.com/php/php-src/blob/221b4fe246e62c5d59f617ee4cbe6fd4c614fb5a/ext/opcache/ZendAccelerator.c#L2800-L2814

Reply:
> opcache doesn't work with embed

thanks for the hint, I updated a few more lines to account for that